irohad/ametsuchi/impl/peer_query_wsv.cpp

```diff
@@ -15,38 +15,38 @@
  * limitations under the License.
  */
 
-#include "builders/protobuf/common_objects/proto_peer_builder.hpp"
-#include "ametsuchi/impl/peer_query_wsv.hpp"
+#include <numeric>
 
+#include "ametsuchi/impl/peer_query_wsv.hpp"
 #include "ametsuchi/wsv_query.hpp"
-
-using wPeer = std::shared_ptr<shared_model::interface::Peer>;
+#include "builders/protobuf/common_objects/proto_peer_builder.hpp"
 
 namespace iroha {
   namespace ametsuchi {
 
     PeerQueryWsv::PeerQueryWsv(std::shared_ptr<WsvQuery> wsv)
         : wsv_(std::move(wsv)) {}
 
-    boost::optional<std::vector<wPeer>> PeerQueryWsv::getLedgerPeers() {
-      const auto &tmp = wsv_->getPeers();
-      if (not tmp) {
-        return boost::none;
-      }
-      std::vector<wPeer> peers = [&tmp] {
-        std::vector<wPeer> result;
-        for (const auto &item : *tmp) {
-          shared_model::proto::PeerBuilder builder;
-
-          auto key = shared_model::crypto::PublicKey(item.pubkey.to_string());
-          auto peer = builder.address(item.address).pubkey(key).build();
-
-          auto curr = std::make_shared<shared_model::proto::Peer>(peer.getTransport());
-          result.emplace_back(curr);
-        }
-        return result;
-      }();
-      return peers;
+    boost::optional<std::vector<PeerQuery::wPeer>>
+    PeerQueryWsv::getLedgerPeers() {
+      return wsv_->getPeers() | [](const auto &peers) {
+        return std::accumulate(
+            peers.begin(),
+            peers.end(),
+            std::vector<wPeer>{},
+            [](auto &vec, const auto &peer) {
+              shared_model::proto::PeerBuilder builder;
+
+              auto key =
+                  shared_model::crypto::PublicKey(peer.pubkey.to_string());
+              auto tmp = builder.address(peer.address).pubkey(key).build();
+
+              auto curr =
+                  std::shared_ptr<shared_model::interface::Peer>(tmp.copy());
+              vec.emplace_back(curr);
```

Comment:
`emplace_back` and `push_back` have the same semantic - copy ctor call, but from the nature of object usage `push_back` is preferable because it is about copy (or move) existed object. I suggest to rework it with `oush_back` or directly pass arguments from `curr` variable.